test/date_range_test.rb

```diff
@@ -0,0 +1,116 @@
+# writting a test for duration
+require_relative "test_helper"
+
+describe "date range class" do
+  it "calculates duration" do
+    # Arrange
+    date_range_instance = DateRange.new("2020-02-27", "2020-03-02")
+    # Act
+
+    duration = date_range_instance.duration()
+
+    # Assert
+    expect (duration).must_equal 4
+  end
+  describe "overlap" do
```

Comment:
These tests are very comprehensive. Nice work! Consider naming them with the particular test case they test for. See the example below from the design scaffold.

Reply:
```
describe "overlap?" do
    before do
      start_date = Date.new(2017, 01, 01)
      end_date = start_date + 3

      @range = Hotel::DateRange.new(start_date, end_date)
    end

    it "returns true for the same range" do
      start_date = @range.start_date
      end_date = @range.end_date
      test_range = Hotel::DateRange.new(start_date, end_date)

      expect(@range.overlap?(test_range)).must_equal true
    end

    xit "returns true for a contained range" do
    end

    xit "returns true for a range that overlaps in front" do
    end

    xit "returns true for a range that overlaps in the back" do
    end

    xit "returns true for a containing range" do
    end

    xit "returns false for a range starting on the end_date date" do
    end

    xit "returns false for a range ending on the start_date date" do
    end

    xit "returns false for a range completely before" do
    end

    xit "returns false for a date completely after" do
    end
  end

  xdescribe "include?" do
    it "reutrns false if the date is clearly out" do
    end

    it "returns true for dates in the range" do
    end

    it "returns false for the end_date date" do
    end
  end
```